### PR TITLE
Add `qnorm`

### PR DESCRIPTION
**Pull request by rikhuijzer**

### CODE REVIEW

File: src/libc.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+pub fn log(x: f64) -> f64 {
+    x.ln()
+}
+
+pub fn sqrt(x: f64) -> f64 {
+    x.sqrt()
+}
```
Comment (rikhuijzer) [inline src/libc.rs:7]: The only drawback to this is that it takes a bit more compilation time since the compiler will hopefully remove this indirection again. However, it makes copy-pasting C code easier so I think it makes sense? @storopoli what do you think?

File: src/dpq.rs [hunk 2]
```diff
@@ -31,3 +34,91 @@ pub fn r_dt_1(lower_tail: bool, log_p: bool) -> f64 {
...
+///
+/// R_Q_P01_check(p);
+/// if (p == R_DT_0) return left;
+/// if (p == R_DT_1) return right;
+///
+/// the following implementation should be more efficient (less tests):
+///
+/// This was originally a macro, but it is now a function.
+/// At the caller site, if the return value is not None, then return the
+/// result immediately.
```
Comment (rikhuijzer) [inline src/dpq.rs:93]: Pay attention with macro's that return. When converting to a function, make sure to propagate the immediate return upwards. I did it here with an `Option`. @storopoli agreed?

File: test-math/src/lib.rs [hunk 3]
```diff
@@ -27,19 +29,37 @@ mod test_math {
...
+        });
+        assert_eq!(qnorm(0.4, 0.5, 1.0, false, false), unsafe {
+            c::qnorm5(0.4, 0.5, 1.0, 0, 0)
+        });
+        assert!(abs_diff_eq!(
+            qnorm(-2.3, 0.5, 1.0, false, true),
+            unsafe { c::qnorm5(-2.3, 0.5, 1.0, 0, 1) },
+            epsilon = 1e-15
+        ));
+    }
```
Comment (storopoli) [inline test-math/src/lib.rs:64]: Can you add tests to catch the `ML_NAN`s return? Like negative sigma's etc.

Comment (rikhuijzer) [reply test-math/src/lib.rs:64]: I had to set `nmath` to `pub` in `src/lib.rs`. Then I could access it via `using rmathlib::nmath::ML_POSINF`.

However, I couldn't access the C macro in Rust. Rust can only access constants and functions from what I could find. See also https://stackoverflow.com/questions/21485655/how-do-i-use-c-preprocessor-macros-with-rusts-ffi

So let's declare checking C macro's as "would have been nice to have", but not easy enough to do? As long as the tests on the top-level functions, such as `qnorm` pass it should be fine.

File: src/qnorm.rs [hunk 4]
```diff
@@ -0,0 +1,159 @@
+use crate::dpq::*;
+use crate::libc::*;
+use crate::nmath::*;
+use crate::rmath::*;
+
+/// Compute the quantile function for the normal distribution.
+///
+/// The algorithm AS 241 of Wichura is used,
+///     and has been improved for the very extreme tail (and log_p=TRUE)
```
Comment (storopoli) [inline src/qnorm.rs:9]:
```suggestion
/// ## DESCRIPTION
///
/// The algorithm AS 241 of Wichura is used,
/// and has been improved for the very extreme tail (and log_p=TRUE)
```

These are tab-spaced ones in the original c

File: src/qnorm.rs [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+///
+/// REFERENCE
```
Comment (storopoli) [inline src/qnorm.rs:11]:
```suggestion
/// ## REFERENCE
```

Let's try to keep like this?
I know that C docstrings are awful, but we could show how to do better?

File: src/qnorm.rs [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    //      tail area of P; Z is accurate to about 1 part in 10**16.
+    //
+    //      (original fortran code used PARAMETER(..) for the coefficients
+    //       and provided hash codes for checking them...)
+    //
+    if q.abs() <= 0.425 {
+        // |p~ - 0.5| <= .425  <==> 0.075 <= p~ <= 0.925
+        r = 0.180625 - q * q; // = .425^2 - q^2  >= 0
+        val = q
+            * (((((((r * 2_509.080_928_730_122_7 + 33_430.575_583_588_13) * r
```
Comment (storopoli) [inline src/qnorm.rs:53]: Just a comment. We are porting LISP 😂

File: test-math/Cargo.toml [hunk 7]
```diff
@@ -7,6 +7,7 @@ edition = "2021"
 cc = "1.0"
 
 [dependencies]
+approx = "0.5.1"
```
Comment (storopoli) [inline test-math/Cargo.toml:10]:
```suggestion
approx = "0.5"
```

I prefer the compat bounds cleaner.

Actually can we move all of this to `dev-dependencies` since we are using for `cargo test`.

If so, adjust the compat of `approx` there as well.

File: test-math/src/lib.rs (hunk 3 above)
Comment (storopoli) [reply test-math/src/lib.rs:64]: Maybe we can only test

```rust
assert!(result.is_nan())
```
?

**Review by storopoli — CHANGES_REQUESTED** [review]

File: test-math/src/lib.rs (hunk 3 above)
Comment (rikhuijzer) [reply test-math/src/lib.rs:64]: I see tests as a declarative way to say "given these inputs, I expect this output". However, for constants such as `ML_NAN`, the code itself is already declarative, so adding a test doesn't really add value.
